File: PREGUNTASYRESPUESTAS/FrontEnd/src/app/models/respuestaCuestionario.ts
import { RepuestaCuestionarioDetalle } from "./respuestaCuestionarioDetalle";

export class respuestaCuestionario{
  id: number ;
  fecha: Date | null;
  activo: number;
  cuestionarioId:number | null;
  nombreParticipante:string;
  listRtaCuestionarioDetalle: RepuestaCuestionarioDetalle[];

  constructor(cuestionarioId: number | null, nombreParticipante: string,
    listRtaCuestionarioDetalle: RepuestaCuestionarioDetalle[],
    id: number ,
    fecha: Date | null,
    activo: number,) {
      this.cuestionarioId = cuestionarioId;
      this.nombreParticipante = nombreParticipante;
      this.listRtaCuestionarioDetalle = listRtaCuestionarioDetalle;
      this.id = id;
      this.fecha = fecha;
      this.activo = activo;
  }

}

File: PREGUNTASYRESPUESTAS/Backend/Backend/Backend/Domain/Models/RespuestaCuestionario.cs
﻿using System.ComponentModel.DataAnnotations.Schema;

namespace Backend.Domain.Models
{
    public class RespuestaCuestionario
    {
        public int Id { get; set; }
        [Column(TypeName = "varchar(100)")]
        public required string NombreParticipante { get; set; }

        public DateTime Fecha { get; set; }

        public int Activo { get; set; }
        public int CuestionarioId { get; set; }

        public Cuestionario Cuestionario { get; set; }

        public List<RespuestaCuestionarioDetalle> ListRtaCuestionarioDetalle { get; set; }
    }
}
